app/models/waste_carriers_engine/registration.rb

```diff
@@ -52,6 +52,10 @@ def generate_renew_token!
       save!
     end
 
+    def already_renewed?
+      check_service.date_can_renew_from > Time.now.in_time_zone("London").to_date
```

Comment:
Won't this also return `true` if the registration is expiring in 4 months from now, and can't be renewed for another month? I feel like this could confuse people.

Reply:
No, I don't think so. 🤔 I have explored different approaches to decide if a registration was already renewed or not. Every one of them was `flaky` in some part. I choose this at the end as it was the one that made more sense. Happy to discuss the approach tomorrow

Reply:
I believe @irisfaraway is right. Using my local environment I created a situation where I had a registration with an `expires_on` of `2020-09-14` and no `past_registrations`. This put it at 4 months before it expired, and 1 month before it could be renewed.

When I called `already_renewed?` I got `true`.

<img width="304" alt="Screenshot 2020-05-14 at 23 14 17" src="https://user-images.githubusercontent.com/1789650/81993127-f7ee1180-963c-11ea-8c9d-a2811821bcbe.png">

To try and be helpful I had a stab at coming up with a solution and came up with this.

```ruby
registration_window = reg.expires_on - Rails.configuration.expires_after.years
reg.past_registrations.where(cause: nil, :expires_on.gte => registration_window ).count
```

So I created another registration, this time with an `expires_on` of `2020-06-14`. With nothing in `past_registrations` the query returned `0`. I then completed a renewal and tried again. This time I got `1`.

I then went in and added another registration to `past_registrations` and set its registered on dates to `2014-06-14` and its `expires_on` to `2017-06-14`. So now I have 2 registrations, one covering 2014 to 2017, and the other 2017 to 2020.

Running `reg.past_registrations.where(cause: nil, :expires_on.gte => registration_window ).count` still returns me 1 as expected. If `past_registrations` only contained the 2014-2017 registration, I would not consider the current registration to have been renewed as the past one is too old. I'd expect the query to return `0` (which it does).

So not necessarily the final solution. But hopefully, something that helps.

Reply:
Well, I like it. And using "nil" to refer to past registrations that are renewal will also remove the confusion on the other PR. Will make the changes